### PR TITLE
Proposal for a bigger access token 

### PR DESCRIPTION
**Pull request by tomdaffurn**
Propose a fatter access token: 

1. Comply fully with RFC9068
    - add `aud`. This is our Issuer, which will probably be the same DID as our Authorization Server, but they are distinct in OAuth
    - add `client_id`. The applicant DID from SIOPv2 request. Will be the same as `sub`, but is required by the spec
    - add `jti`
2. Add `c_nonce` abd `c_nonce_expiry`. These are included so the Credential Endpoint / Issuer can consume the token without calling back to the Authorization server. It's fewer fields in our DB. They're not part of RFC9068, but are endorsed by it: 
> in-market use has shown that many commercial OAuth 2.0 implementations elected to issue access tokens using a format that can be parsed and validated by resource servers directly, without further authorization server involvement.

> [Use] JWT claims to convey the information needed to support a common set of use cases: token validation, transporting authorization information in the form of scopes and entitlements, carrying identity information about the subject, and so on

### CODE REVIEW

**Review by mistermoe — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -521,22 +524,27 @@ Clients must use the fields from token response in subsequent calls to the [3. I
...
-| `iat` | (IssuedAt) The time at which the `access_token` was granted        | y        | [RFC7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.6) |          |
+| Field            | Description                                                          | Required | References                                                                                           | Comments |
+| :--------------- | :------------------------------------------------------------------- | :------- | :--------------------------------------------------------------------------------------------------- | :------- |
+| `iss`            | (Issuer) The DID of the Authorization Server                         | y        | [RFC7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.1)                               |          |
+| `aud`            | (Audience) The Resource Identifier of the Credential Issuer          | y        | [RFC7519](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3)                               |          |
```
Comment (decentralgabe) [inline README.md:540]: by resource identifier do you mean the issuer's DID?

**Review by decentralgabe — APPROVED** [review]
lgtm!

**Comment by KendallWeihe** [comment]
> in-market use has shown that many commercial OAuth 2.0 implementations elected to issue access tokens using a format that can be parsed and validated by resource servers directly, without further authorization server involvement

So is the idea that the credential issuer will validate the proof jwt's nonce against the access token's nonce? 🤔 yeah I like it! nice!

**Review by KendallWeihe — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (tomdaffurn) [reply README.md:540]: yep

**Comment by tomdaffurn** [comment]
> did we have a concrete use case for the addition of `jti`? I'm presuming it's an indexed column in the db for lookups?

No strong need for it, but it's REQUIRED in RFC9068. It definitely can be a DB ID. It also makes each token unique

**Comment by EbonyLouis** [comment]
@tomdaffurn @KendallWeihe @decentralgabe @mistermoe Hey guys for any future KCC changes please add me to the conversation, I don't want our guides to ever become outdated/misleading. - thanks!

**Comment by decentralgabe** [comment]
hey @EbonyLouis I would recommend 'watch' ing the repo so you auto get notifications
I will also add you as a codeowner

**Comment by EbonyLouis** [comment]
Great idea, thanks @decentralgabe !

**Pull request merged**